site/content/integrate/apps/functionality/kv-store.md

```diff
@@ -4,19 +4,99 @@ description: Use the per-App key-value data store
 weight: 20
 ---
 The Apps framework provides access to a simple key-value store for App-specific data.
-Keys are simple strings and values are any type that can be marshalled to JSON.
+Keys are simple strings and values are any data that can be expressed in JSON.
 
 {{<note "Note:">}}
-It is recommended to use a [driver]({{<ref "/integrate/apps/drivers">}}) when using the key-value store.
+It is not recommended to store App configuration in the key-value store as an access token, either a user or bot token, is required to access it.
```

Comment:
I don't understand what this means. The App will have access to the user or bot token when it wants to access this configuration right? For HTTP-based apps, they can store that information elsewhere if they want to. In the case where something occurs outside of the ecosystem of Mattermost (such as webhook call, or some other non-Mattermost event/trigger), the App framework's webhook endpoint should be used to "poke" the App with the bot token.

I suppose my question is, what is this "App configuration" being mentioned here, that would in some scenario make since to _not_ store in the KV store?

Reply:
Stating the fact that an appropriate access token is needed seems to be good, though I think this is mentioned elsewhere when we talk about the user-based scoping in the store. Yeah I think we can remove this note